swarm-derive/src/lib.rs

```diff
@@ -70,72 +70,42 @@ fn build_struct(ast: &DeriveInput, data_struct: &DataStruct) -> TokenStream {
         quote! {<#(#lf,)* #(#tp,)* #(#cst,)*>}
     };
 
-    // Whether or not we require the `NetworkBehaviourEventProcess` trait to be implemented.
-    let event_process = {
-        let mut event_process = false;
-
-        for meta_items in ast.attrs.iter().filter_map(get_meta_items) {
-            for meta_item in meta_items {
-                match meta_item {
-                    syn::NestedMeta::Meta(syn::Meta::NameValue(ref m))
-                        if m.path.is_ident("event_process") =>
-                    {
-                        if let syn::Lit::Bool(ref b) = m.lit {
-                            event_process = b.value
-                        }
-                    }
-                    _ => (),
-                }
-            }
-        }
-
-        event_process
-    };
-
     // The fields of the struct we are interested in (no ignored fields).
     let data_struct_fields = data_struct
         .fields
         .iter()
         .filter(|f| !is_ignored(f))
         .collect::<Vec<_>>();
 
-    // The final out event.
-    // If we find a `#[behaviour(out_event = "Foo")]` attribute on the struct, we set `Foo` as
-    // the out event. Otherwise we use `()`.
-    let out_event = {
-        let mut out = quote! {()};
-        for meta_items in ast.attrs.iter().filter_map(get_meta_items) {
-            for meta_item in meta_items {
-                match meta_item {
-                    syn::NestedMeta::Meta(syn::Meta::NameValue(ref m))
-                        if m.path.is_ident("out_event") =>
-                    {
-                        if let syn::Lit::Str(ref s) = m.lit {
-                            let ident: syn::Type = syn::parse_str(&s.value()).unwrap();
-                            out = quote! {#ident};
-                        }
-                    }
-                    _ => (),
-                }
+    let event = {
```

Comment:
Why not at least give the user the option to set their own `OutEvent`? Per default it could still be generated. This would also avoid breaking the implementation of current users that already use a custom `OutEvent`.

Reply:
In which use-case would a user need their own `OutEvent`? Asked the other way around, which use-case would not be supported with a generated `OutEvent`?

Reply:
I don't think they strictly _need_ their own `OutEvent`. But it may be more convenient. Especially because of the point that @thomaseizinger raised above, that is the interaction with a generated enum. As a user I'd always prefer to have my own type rather than one generated by a macro. Most likely I would parse the OutEvent into my custom type anyway, e.g. when bubbling the event up to a higher layer. So still having to deal with the generated one would probably just be extra work (understand how the enum looks like, where the variant names come from, handling changes in the enum if this variant names change because a field was renames, etc).